### PR TITLE
refactor: switch from xsalsa20 to chacha20

### LINKED ISSUE #294: Change default encryption to x25519-chacha20-poly1305

### LINKED ISSUE #339: Use padding for encryption

### PR DESCRIPTION
**Pull request by hai-ko**
closes #294
closes #339

### CODE REVIEW

File: packages/lib/src/delivery/Messages.ts [hunk 1]
```diff
@@ -86,46 +88,39 @@ export async function incomingMessage(
...
-    return hexlify(toUtf8Bytes(encryptedPostmark));
+    };
 }
 
-const signPostmark = (p: Omit<Postmark, 'signature'>, signingKey: string) => {
+function signPostmark(
+    p: Omit<Postmark, 'signature'>,
+    signingKey: string,
+): Promise<string> {
+    log(`SING POST MARK USING ${signingKey}`);
```
Comment (AlexNi245) [inline packages/lib/src/delivery/Messages.ts:123]: Do we want to keep this log statement? I think it's probably left from one of my PR's maybe we can remove It now.

File: packages/lib/src/account/Account.ts [hunk 2]
```diff
@@ -31,22 +28,20 @@ import { UserProfileResolver } from './profileResolver/json/UserProfileResolver'
...
-}
-
 export interface UserProfile {
     publicEncryptionKey: string;
     publicSigningKey: string;
     deliveryServices: string[];
     mutableProfileExtensionUrl?: string;
 }
 
+export interface Keys {
```
Comment (AlexNi245) [inline packages/lib/src/account/Account.ts:38]: Wouldnt it makes sense to move this interface to the KeyCreation.ts file?

File: packages/lib/src/session/SignIn.ts [hunk 3]
```diff
@@ -53,17 +67,19 @@ export async function signIn(
...
-                connection,
-                personalSign,
-                getSymmetricalKeyFromSignature,
-            );
+        const keys = await createKeys(
+            await createStorageKey(signedNonceMsg),
+            nonce,
+        );
 
+        if (!externalDataFile && !browserDataFile) {
```
Comment (AlexNi245) [inline packages/lib/src/session/SignIn.ts:79]: Is this just happen during development? Making a comment here would make sense to explain this code section.

File: packages/react/src/sign-in/Connectors.ts [hunk 4]
```diff
@@ -169,9 +168,11 @@ export async function signIn(
                 break;
 
             case Lib.storage.StorageLocation.dm3Storage:
-                let authToken = (await localforage.getItem(
-                    'ENS_MAIL_AUTH_' + account.address,
-                )) as string;
+                let authToken = null;
+                // (await localforage.getItem(
```
Comment (AlexNi245) [inline packages/react/src/sign-in/Connectors.ts:172]: Why is this code commented out?

File: packages/react/src/utils/SigCheck.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    accountAddress: string,
+    signature: string,
+): Promise<boolean> {
+    const isValid = await Lib.crypto.checkSignature(
+        publicSigningKey,
+        stringify(message)!,
+        signature,
+    );
+
+    if (!isValid) {
```
Comment (AlexNi245) [inline packages/react/src/utils/SigCheck.ts:16]: Why not return false here already? After this there is no way the function can return true anymore.

**Review by AlexNi245 — CHANGES_REQUESTED** [review]
See comments

File: packages/react/src/utils/SigCheck.ts (hunk 5 above)
Comment (hai-ko) [reply packages/react/src/utils/SigCheck.ts:16]: Yes, I simplified the function

File: packages/lib/src/session/SignIn.ts (hunk 3 above)
Comment (hai-ko) [reply packages/lib/src/session/SignIn.ts:79]: There will be an issue addressing the sing in / auth.

**Review by AlexNi245 — APPROVED** [review]

**Pull request merged**